contracts/utility/Upgradeable.sol

```diff
@@ -83,6 +83,9 @@ abstract contract Upgradeable is IUpgradeable, AccessControlEnumerableUpgradeabl
      */
     function postUpgrade(bytes calldata data) external {
         uint16 initializations = _initializations + 1;
+        if (initializations < version()) {
```

Comment:
The idea was preventing accidental non-incremental version upgrades. The case that you have version 5 and trying to upgrade it to version 8 shouldn't happen, no?

Reply:
What we're trying to prevent here is someone calling postUpgrade multiple times.

We have multiple different scenarios:
1. CarbonController is deployed with version 2 without the proxy delegate check, and then immediately upgraded with the proxy delegate check (postUpgrade is called)
2. Other contracts are deployed with version 1 and no upgrade (postUpgrade isn't called)
3. Multiple different versions are deployed to different chains, so we don't have to maintain a version number per chain, this can result in a deployment of version 3 to chain X